Prox/Prox/PlaceDetails/PlaceDetailViewController.swift

```diff
@@ -97,52 +110,242 @@ class PlaceDetailViewController: UIViewController {
         // Dispose of any resources that can be recreated.
     }
 
+    fileprivate func dequeuePlaceCardViewController(forPlace place: Place) -> PlaceDetailsCardViewController {
+        if let placeCardVC = unusedPlaceCardViewControllers.last {
+            unusedPlaceCardViewControllers.removeLast()
+            placeCardVC.place = place
+            return placeCardVC
+        }
 
-    func pageControlDidPage(sender: AnyObject) {
-        let pageSize = imageCarousel.bounds.size
-        let xOffset = pageSize.width * CGFloat(pageControl.currentPage)
-        imageCarousel.setContentOffset(CGPoint(x: xOffset, y: 0), animated: true)
+        return PlaceDetailsCardViewController(place: place)
     }
-    
-}
 
-extension PlaceDetailViewController: UICollectionViewDataSource {
-    func numberOfSections(in collectionView: UICollectionView) -> Int {
-        return 1
+    var startConstant: CGFloat!
+
+    func didPan(gestureRecognizer: UIPanGestureRecognizer) {
+
+        if gestureRecognizer.state == .began {
+            startConstant = currentCardViewCenterXConstraint?.constant ?? 0
```

Comment:
If we don't have a `currentCardViewCenterXConstraint`, something has gone wrong, right? I think you've made the case that the UI should degrade gracefully with missing data but perhaps we should throw rather than mask an unexpected state (here & below).